### PR TITLE
fully separate core metrics

### PR DESCRIPTION
**Pull request by KatKatKateryna**

### CODE REVIEW

**Comment by gjedlicska** [comment]
Lets try to follow PEP8 in naming and style conventions: https://peps.python.org/pep-0008/

Class names should be CapitalCase, eg `Core_Resource` should be `CoreResource`

File: src/specklepy/api/host_applications.py [hunk 1]
```diff
@@ -2,34 +2,15 @@
...
+from specklepy.core.api.host_applications import (HostAppVersion,
+                                                  RHINO,GRASSHOPPER,REVIT,DYNAMO,UNITY,GSA,
+                                                  CIVIL,AUTOCAD,MICROSTATION,OPENROADS,
+                                                  OPENRAIL,OPENBUILDINGS,ETABS,SAP2000,CSIBRIDGE,
+                                                  SAFE,TEKLASTRUCTURES,DXF,EXCEL,UNREAL,POWERBI,
+                                                  BLENDER,QGIS,ARCGIS,SKETCHUP,ARCHICAD,TOPSOLID,
+                                                  PYTHON,NET,OTHER)
 
 @dataclass
 class HostApplication:
```
Comment (gjedlicska) [inline src/specklepy/api/host_applications.py:16]: This definition can be a re-export from core too

File: src/specklepy/api/host_applications.py [hunk 2]
```diff
@@ -103,7 +52,6 @@ def get_version(self, version: HostAppVersion) -> str:
     "net": NET,
 }
 
-
 def get_host_app_from_string(app_name: str) -> HostApplication:
```
Comment (gjedlicska) [inline src/specklepy/api/host_applications.py:55]: this is a re-export too

File: src/specklepy/api/host_applications.py [hunk 3]
```diff
@@ -39,38 +20,6 @@ class HostApplication:
...
-QGIS = HostApplication("QGIS", "qgis")
-ARCGIS = HostApplication("ArcGIS", "arcgis")
-SKETCHUP = HostApplication("SketchUp", "sketchup")
-ARCHICAD = HostApplication("Archicad", "archicad")
-TOPSOLID = HostApplication("TopSolid", "topsolid")
-PYTHON = HostApplication("Python", "python")
-NET = HostApplication(".NET", "net")
-OTHER = HostApplication("Other", "other")
-
 _app_name_host_app_mapping = {
```
Comment (gjedlicska) [inline src/specklepy/api/host_applications.py:23]: this is prob not needed after importing everything from core

File: src/specklepy/api/credentials.py [hunk 4]
```diff
@@ -11,9 +11,35 @@
...
-from specklepy.core.api.credentials import (UserInfo, Account, StreamWrapper, 
-                                            get_account_from_token,
-                                            get_local_accounts as core_get_local_accounts)
+from specklepy.core.api.credentials import (UserInfo, 
+                                            StreamWrapper, # deprecated 
+                                            get_local_accounts as core_get_local_accounts, 
+                                            get_account_from_token as core_get_account_from_token)
+
+
+class Account(BaseModel):
```
Comment (gjedlicska) [inline src/specklepy/api/credentials.py:20]: This can be imported from core

File: src/specklepy/api/resources/stream.py [hunk 5]
```diff
@@ -144,7 +138,6 @@ def favorite(self, stream_id: str, favorited: bool = True):
             Stream -- the stream with its `id`, `name`, and `favoritedDate`
         """
         metrics.track(metrics.SDK, self.account, {"name": "Stream Favorite"})
-
         return super().favorite(stream_id, favorited)
 
     @deprecated(
```
Comment (gjedlicska) [inline src/specklepy/api/resources/stream.py:143]: lets delete this definition. It has been deprecated for a while.

File: src/specklepy/api/resources/user.py [hunk 6]
```diff
@@ -18,15 +19,14 @@
 )
 
 
-class Resource(ResourceBase):
+class Resource(Core_Resource):
```
Comment (gjedlicska) [inline src/specklepy/api/resources/user.py:22]: Lets delete this class and module, it's been deprecated for a while now

**Review by gjedlicska — COMMENTED** [review]
I've left some notes, mostly stylistic changes. The important parts look good

**Review by gjedlicska — CHANGES_REQUESTED** [review]
Lets make the few noted changes

**Comment by KatKatKateryna** [comment]
@gjedlicska changes applied except the 'user' resource - will be deprecated in 2.16

**Review by gjedlicska — APPROVED** [review]

**Pull request merged**